tools/plugins/com.liferay.ide.project.ui/src/com/liferay/ide/project/ui/dialog/LiferayServiceProjectSelectionDialog.java

```diff
@@ -54,8 +53,22 @@ protected boolean checkProject( IJavaProject javaProject )
         }
     }
 
-    public void setProjects( List<IProject> projects )
+    private List<IProject> getServiceBuilderProjects()
     {
-        this.projects = projects;
+        List<IProject> results = new ArrayList<IProject>();
+
+        IProject[] projects = CoreUtil.getAllProjects();
+
+        for( IProject project : projects )
+        {
+            IFile serviceFile = project.getFile( "/docroot/WEB-INF/service.xml" );
```

Comment:
Yeah both of those "actions" shouldu go in the `IProjectBuilder` api so we can do it in both PluginSDK (ant) and maven